research/json-crdt.md

```diff
@@ -0,0 +1,134 @@
+# JSON-CRDT
+
+This document explains the internals of JSON-CRDTs based on the 
+[Conflict-Free weplicated JSON Datatype](https://arxiv.org/pdf/1608.03960.pdf) 
+paper.
+
+JSON-CRDT is a distributed data type which automatically resolves concurrent 
+modifications in the JSON structure without losing any update. The JSON-CRDT
+provide strong eventual consistency and does not depend on ordering guarantees
+from the network.
+
+These properties make the JSON-CRDT a good fit for p2p applications or any
+state-synchronizing application.
+
+## Types
+
+### MapT
+### ListT
+### RegisterT
+
+## Editing the JSON document and replica local state
+
+Each replica keeps its own state of the document. Changes in the document 
+occur only locally. When the document is edited, it generates an `operation` 
+which uniquely describes the mutation globally. The operations are distributed 
+to all peers as a mechanism to reach state consistency across all peers.
+
+Once a replica receives an external operation, it applies the operation locally. 
+The merging algorithm ensures that applying external operations locally will 
+not conflict with the current local state.
+
+The state of the document at any point in time is a set of operations which were
+applied locally.
+
+## Operations
+
+### Operation Representation
+
+An operation is a tuple of the form
+
+```
```

Comment:
Maybe using a screenshot of the paper where the tuple is represented would be easier to read. Since this is really informal, I believe we can just do it and loosely reference it?

Reply:
Yes, screenshot plus ref at the end is fine.